app/controllers/assets_controller.rb

```diff
@@ -38,8 +38,9 @@ def restrict_request_format
   end
 
   def asset_params
+    asset_params = handle_empty_access_limited_param(params)
     exclude_blank_redirect_url(
-      params
+      asset_params
         .require(:asset)
         .permit(:file, :draft, :redirect_url, :replacement_id, :parent_document_url, access_limited: [])
     )
```

Comment:
A possible alternative approach which I think looks a bit cleaner:

```ruby
  def asset_params
      exclude_blank_redirect_url(
        handle_empty_access_limited_param(params)
          .require(:asset)
          .permit(:file, :draft, :redirect_url, :replacement_id, :parent_document_url, access_limited: [])
      )
    )
  end
```

Reply:
I've changed this in a new commit in this branch.